### PR TITLE
LibWeb: Handle inline-start and inline-end as float values

### LINKED ISSUE #449: float: inline-end doesn't work
My understanding is that `inline-end` is like `right`, but flips in RTL languages.

Issue observed in this example:

```html
<!DOCTYPE html>
<html lang="en">
	<body>
		<p style="float: right">right</p>
		<p>none</p>
		<p style="float: inline-end">inline-end</p>
	</body>
</html>
```

Shows in Firefox and Epiphany as:
Screen Shot 2024-07-05 at 22 13 44

But in Ladybird as:
screenshot-2024-07-05-22-16-16

The inspector shows the computed float value as "none", so it seems to be ignoring the value. Why the second right-aligned paragraph is offset I have no idea, but it seems to happen with `right` too

### PR DESCRIPTION
**Pull request by vpzomtrrfrt**
Should resolve #449 for LTR languages at least (as far as I can tell, writing-mode isn't supported at all anyway?)

### CODE REVIEW

File: Userland/Libraries/LibWeb/Layout/BlockFormattingContext.cpp [hunk 1]
```diff
@@ -1077,9 +1077,19 @@ void BlockFormattingContext::layout_floating_box(Box const& box, BlockContainer
...
-    if (box.computed_values().float_() == CSS::Float::Left) {
+
+    Optional<FloatSide> float_side;
+
+    // FIXME: Honor writing-mode, direction and text-orientation.
+    if (box.computed_values().float_() == CSS::Float::Left || box.computed_values().float_() == CSS::Float::InlineStart) {
+        float_side = FloatSide::Left;
+    } else if (box.computed_values().float_() == CSS::Float::Right || box.computed_values().float_() == CSS::Float::InlineEnd) {
+        float_side = FloatSide::Right;
+    }
```
Comment (awesomekling) [inline Userland/Libraries/LibWeb/Layout/BlockFormattingContext.cpp:1088]: We resolve other direction-dependent values in StyleComputer, i.e before layout even happens. I think that could work for these as well :)

Comment (vpzomtrrfrt) [reply Userland/Libraries/LibWeb/Layout/BlockFormattingContext.cpp:1088]: Does that mean defining a new StyleValue subclass?

Comment (awesomekling) [reply Userland/Libraries/LibWeb/Layout/BlockFormattingContext.cpp:1088]: That should not be necessary. We just need `StyleComputer::set_property_expanding_shorthands()` to somehow turn `float: inline-end` into `float: right` or whatever. :)

If you look around that file, you can see how other logical properties are handled, for example in `for_each_property_expanding_shorthands()` (although note that those are logical properties, not logical values)

Comment (vpzomtrrfrt) [reply Userland/Libraries/LibWeb/Layout/BlockFormattingContext.cpp]: ok I've moved it there

**Comment by AtkinsSJ** [comment]
Changes look good, but please combine the two commits together, because the second just corrects the first.

**Pull request merged**